backend/app/usecase/match/keyword.go

```diff
@@ -0,0 +1,32 @@
+package match
+
+import (
+	"fmt"
+)
+
+// KeywordType represents keyword matching type.
+type KeywordType int
+
+const (
+	// ContainsAllKeywords represents ContainsAll type
+	ContainsAllKeywords KeywordType = iota
+	// ContainsAnyKeyword represents ContainsAny type
+	ContainsAnyKeyword
+)
+
+// Keyword matches a list of words against an input.
+type Keyword interface {
```

Comment:
Just curious if `Keyword` is the right term. Shouldn't it have some sort of `Matcher` in the name?

Reply:
The package name is match, so the term here is match.Keyword. If we name it something like match.KeywordMatcher it seems too much unnecessary information. I agree however that it is a bit hard to follow. I will modify it now.

Reply:
Okay, understood.

Reply:
Let's rename the package to matcher. We want to reuse this package for future input validation also. If both contains all and contain any are changed to standalone function, this file is not needed any more.

Reply:
I had it in the begging named matcher, but was in dilemma 😅